src/flint/fq_default_mpoly.jl

```diff
@@ -27,7 +27,7 @@ function ordering(a::FqMPolyRing)
 end
 
 function gens(R::FqMPolyRing)
-    return [FqMPolyRingElem(R, a) for a in gens(R.data)]
+    return FqMPolyRingElem[FqMPolyRingElem(R, a) for a in gens(R.data)]
```

Comment:
This potentially causes extra allocations. Efficient variant:
```suggestion
    return[FqMPolyRingElem(R, a):: FqMPolyRingElem for a in gens(R.data)]
```

To demonstrate the difference with an artificial example:
```
julia> v = Any[1,2,3,4,5,6];

julia> @benchmark [x for x in v]
BenchmarkTools.Trial: 10000 samples with 285 evaluations.
 Range (min … max):  279.968 ns …  1.533 μs  ┊ GC (min … max): 0.00% … 73.11%
 Time  (median):     287.428 ns              ┊ GC (median):    0.00%
 Time  (mean ± σ):   291.021 ns ± 35.212 ns  ┊ GC (mean ± σ):  0.35% ±  2.33%

       ▂▇█▆
  ▂▂▂▃▆█████▇▆▄▄▃▃▄▄▄▃▃▃▃▃▂▂▂▂▂▂▂▂▂▂▂▂▂▂▂▂▂▂▂▂▂▂▂▂▂▂▂▁▂▂▂▂▂▂▂▂ ▃
  280 ns          Histogram: frequency by time          335 ns <

 Memory estimate: 160 bytes, allocs estimate: 4.
```
The memory usage doubles if we force the `Vector{Int}` via prefixing `Int`:
```
julia> @benchmark Int[x for x in v]
BenchmarkTools.Trial: 10000 samples with 66 evaluations.
 Range (min … max):  851.636 ns …  14.352 μs  ┊ GC (min … max): 0.00% … 90.50%
 Time  (median):     861.742 ns               ┊ GC (median):    0.00%
 Time  (mean ± σ):   874.252 ns ± 244.345 ns  ┊ GC (mean ± σ):  0.53% ±  1.79%

   ▄▆█▇▇▆▅▄▃▃▂▂▂▂▂▃▂▂▂▂▁▂▂▂▁▁▁                                  ▂
  ███████████████████████████████▇▆▇▇▇▇█▆▇▆▆▅▆▆▅▃▆▆▅▅▅▄▅▆▅▅▁▄▃▅ █
  852 ns        Histogram: log(frequency) by time        979 ns <

 Memory estimate: 320 bytes, allocs estimate: 8.
```
If we instead use a type assertion, we get type stable and more efficient code:
```
julia> @benchmark [x::Int for x in v]
BenchmarkTools.Trial: 10000 samples with 746 evaluations.
 Range (min … max):  170.633 ns … 609.920 ns  ┊ GC (min … max): 0.00% … 65.87%
 Time  (median):     172.141 ns               ┊ GC (median):    0.00%
 Time  (mean ± σ):   174.291 ns ±  17.860 ns  ┊ GC (mean ± σ):  0.41% ±  2.88%

  ▁▄▇█▇▆▆▅▄▃▂▁ ▂▃▃▂▂▂▂▁▂▁▁▁                                     ▂
  ████████████████████████████▇▇▇▇▇▇▇▇▇▇▇▆▇▆▅▆▅▆▆▆▅▅▅▅▃▄▄▂▄▄▂▃▄ █
  171 ns        Histogram: log(frequency) by time        191 ns <

 Memory estimate: 128 bytes, allocs estimate: 2.
```